src/serialport_unix.cpp

```diff
@@ -131,6 +131,9 @@ void EIO_Open(uv_work_t* req) {
   OpenBaton* data = static_cast<OpenBaton*>(req->data);
 
   int flags = (O_RDWR | O_NOCTTY | O_NONBLOCK | O_CLOEXEC | O_SYNC);
+  if(data->hupcl == false) {

```

Comment:
This is a good patch and I'll incorporate it into my fork but this block should be removed, not moved.  `HUPCL` is not an fcntl flag; it's a tty control flag and has nothing to do with open(2).  On Linux x86_64 HUPCL happens to have the same value as O_APPEND so is benign but on other platforms this code might change the intended open behavior.